File: win-prog/ProgWinVB/Taming the Mouse/EvenBetterBlockOut/CaptureLossNotifyWindow.vb
'--------------------------------------------------------
' CaptureLossNotifyWindow.vb (c) 2002 by Charles Petzold
'--------------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Interface ICaptureLossNotify
    Sub OnLostCapture()
End Interface

Class CaptureLossNotifyWindow
    Inherits NativeWindow

    Public ctrl As ICaptureLossNotify

    Protected Overrides Sub WndProc(ByRef message As Message)
        ' Process WM_CAPTURECHANGED message.
        If message.Msg = 533 Then ctrl.OnLostCapture()
        MyBase.WndProc(message)
    End Sub
End Class

File: win-prog/ProgWinCS/Taming the Mouse/EvenBetterBlockOut/CaptureLossNotifyWindow.cs
//------------------------------------------------------
// CaptureLossNotifyWindow.cs © 2001 by Charles Petzold
//------------------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

interface ICaptureLossNotify
{
     void OnLostCapture();
}

class CaptureLossNotifyWindow: NativeWindow 
{
     public ICaptureLossNotify control;

     protected override void WndProc(ref Message message) 
     {
          if (message.Msg == 533)                 // WM_CAPTURECHANGED
               control.OnLostCapture();

          base.WndProc(ref message);
     }
}
